test/integration/etcd/etcd_storage_path_test.go

```diff
@@ -705,22 +706,22 @@ func startRealMasterOrDie(t *testing.T, certDir string) (*allClient, clientv3.KV
 		}()
 
 		for {
+			listener, port, err := genericapiserveroptions.CreateListener("tcp", "127.0.0.1:0")
+			if err != nil {
+				t.Fatal(err)
+			}
+
 			kubeAPIServerOptions := options.NewServerRunOptions()
+
+			kubeAPIServerOptions.SecureServing.BindPort = port
 			kubeAPIServerOptions.SecureServing.BindAddress = net.ParseIP("127.0.0.1")
+			kubeAPIServerOptions.SecureServing.Listener = listener
```

Comment:
:+1:

Can't we get rid of the whole retry loop with this? @enj

Reply:
Sure, I missed it.